cas-server-core-audit/src/main/java/org/jasig/cas/audit/spi/TicketOrCredentialPrincipalResolver.java

```diff
@@ -115,4 +119,15 @@ private String resolveArgument(final Object arg1) {
         LOGGER.debug("Unable to determine the audit argument. Returning [{}]", UNKNOWN_USER);
         return UNKNOWN_USER;
     }
+
+    /**
+     * Default implementation that simply returns principal#id.
+     */
+    public static class DefaultPrincipalIdProvider implements PrincipalIdProvider {
```

Comment:
Public on purpose - for subclassing purposes. For example custom impl. checks a certain condition/metadata for custom id token and if not met, simply calls 'super.getPrincipalIdFor(..)'